.ci/docker/common/install_executorch.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -ex
+
+source "$(dirname "${BASH_SOURCE[0]}")/common_utils.sh"
+
+clone_executorch() {
+  EXECUTORCH_PINNED_COMMIT=$(get_pinned_commit executorch)
+
+  # Clone the Executorch
+  git clone https://github.com/pytorch/executorch.git
+
+  # and fetch the target commit
+  pushd executorch
+  git checkout "${EXECUTORCH_PINNED_COMMIT}"
+  git submodule update --init
+  popd
+
+  chown -R jenkins executorch
+}
+
+install_buck2() {
+  pushd executorch/.ci/docker
+
+  BUCK2_VERSION=$(cat ci_commit_pins/buck2.txt)
+  source common/install_buck.sh
+
+  popd
+}
+
+install_conda_dependencies() {
+  pushd executorch/.ci/docker
+  # Install conda dependencies like flatbuffer
+  conda_install --file conda-env-ci.txt
+  popd
+}
+
+install_pip_dependencies() {
+  pushd executorch/.ci/docker
+
+  # Install all Python dependencies
+  pip_install -r requirements-ci.txt
+  # NB: Just need to install the stable version of these domain libraries here
+  # so that we can install executorch.  They will need to be recompiled later
+  # together with torch
+  pip_install torchvision torchaudio
```

Comment:
What are the failures if we do not install torchvision and torchaudio here? It looks like those stable versions won't be  used and there is no guarantee that they are compatible with the torch.

Reply:
Here is the import error https://github.com/pytorch/vision/issues/8101.  Basically, it means that you either use prebuilt nightly binaries for torch, vision, and audio or you need to build all of them from source.  Otherwise, there will be import error when trying to import torchvision.  This means that ET will also need to pin commits not only for torch, but also for vision and audio.  Mixing torch pin commit with nightly vision and audio pre-build binaries won't work

Reply:
> lol, sorry. I mean the above answer for the question below when I pip install vision and audio in test.sh.
> 
> For this question, may be I don't really need this step. I add them only for `pip_install .` ExecuTorch later to install the python code. Do you know if that step require any version of vision or audio? I think I will get rid of this part to see if CI still works to confirm

Yeah, the error shows that torchvision and torchaudio are needed by torch not ExecuTorch. And it's only needed when build and run something. As far as I know, as deps to ExecuTorch, torchvision and torchaudio are needed by running example models only, so fetching and building torchvision and torchaudio right before running the ET tests would be ideal.